pkg/cli/admin/release/extract_tools.go

```diff
@@ -624,22 +623,35 @@ func (o *ExtractOptions) extractCommand(command string) error {
 }
 
 const (
-	// binaryReplacement is the location within the installer binary that we can insert our
-	// release image info string
-	binaryReplacement = "\x00_RELEASE_IMAGE_LOCATION_\x00XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX\x00"
+	// exactReleaseImageBinaryReplacement is the location within the installer binary that we can insert our release image info string
+	exactReleaseImageBinaryReplacement = "\x00_RELEASE_IMAGE_LOCATION_\x00XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX\x00"
+	// preferredNameBinaryReplacement is the location within a binary (oc) that we can insert our release image name string
+	preferredNameBinaryReplacement = "\x00_RELEASE_IMAGE_LOCATION_PREFERRED_NAME_\x00XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX\x00"
 )
 
 // copyAndReplaceReleaseInfo performs a targeted replacement for binaries that contain a special marker string
 // as a constant, replacing the marker with releaseInfo and a NUL terminating byte. It returns true if the
 // replacement was performed.
 func copyAndReplaceReleaseInfo(w io.Writer, r io.Reader, bufferSize int, releaseInfo string) (bool, error) {
-	if len(releaseInfo)+1 > len(binaryReplacement) {
+	// try for exactReleaseImageBinaryReplacement first, for backwards compatibility
+	if len(releaseInfo)+1 > len(exactReleaseImageBinaryReplacement) {
 		return false, fmt.Errorf("the release image pull spec is longer than the maximum replacement length for the binary")
 	}
-	if bufferSize < len(binaryReplacement) {
-		return false, fmt.Errorf("the buffer size must be greater than %d bytes", len(binaryReplacement))
+	if bufferSize < len(exactReleaseImageBinaryReplacement) {
+		return false, fmt.Errorf("the buffer size must be greater than %d bytes", len(exactReleaseImageBinaryReplacement))
 	}
+	matched, err := copyReplace(w, r, releaseInfo, exactReleaseImageBinaryReplacement, bufferSize)
```

Comment:
Calling this twice is not appropriate for very large binaries.  These two code paths (version and image) should be split into separate functions.

Reply:
i know.. was doing that now..